packages/client/utils/getTokenFromSSO.ts

```diff
@@ -1,30 +1,52 @@
 import getOAuthPopupFeatures from './getOAuthPopupFeatures'
+import {GA4SignUpEventEmissionRequiredArgs} from './handleSuccessfulLogin'
 
-const getTokenFromSSO = (url: string) => {
+type SucessReturnType = {
+  token: string
+  ga4Args: GA4SignUpEventEmissionRequiredArgs
+}
+
+type ErrorReturnType = {
+  error: string
+}
+
+type ReturnType = SucessReturnType | ErrorReturnType
+
+const getTokenFromSSO = (url: string): ReturnType | Promise<ReturnType> => {
   // It's possible we prematurely opened a popup named SSO at the URL about:blank to avoid popup blockers
   // Calling window.open again will get a reference to that popup
   // Then, we can update the href to the valid URL
   const popup = window.open(url, 'SSO', getOAuthPopupFeatures({width: 385, height: 550, top: 64}))
-  if (!popup) return {token: null, error: 'Failed to open login popup'}
+  if (!popup) return {error: 'Failed to open login popup'}
   popup.location.href = url
   let closeCheckerId: undefined | number
-  return new Promise<{token: string | null; error: string | null}>((resolve) => {
+  return new Promise<ReturnType>((resolve) => {
     const handler = (event: MessageEvent) => {
       // an extension posted to the opener
       if (typeof event.data !== 'object') return
       const {token, error} = event.data
       if (!token && !error) return
       if (event.origin !== window.location.origin) return
 
+      const params = new URLSearchParams(popup.location.search)
+      const userId = params.get('userId')
+      const isNewUser = params.get('isNewUser') === 'true'
+      const isPatient0 = params.get('isPatient0') === 'true'
+
       window.clearInterval(closeCheckerId)
       popup?.close()
       window.removeEventListener('message', handler)
-      resolve({token, error})
+
+      if (userId) {
+        resolve({token, ga4Args: {userId, isNewUser, isPatient0}})
+      } else {
+        resolve({error: 'userId is null'})
```

Comment:
we're swallowing the useful error from the server here, I think that's why we've been getting some extra support tickets around this message